internal/config/client.go

```diff
@@ -94,6 +97,16 @@ func (c readOnlyClient) RESTMapper() meta.RESTMapper {
 	return c.client.RESTMapper()
 }
 
+func (c readOnlyClient) GroupVersionKindFor(obj runtime.Object) (schema.GroupVersionKind, error) {
+	// FIXME: Broken implementation.
```

Comment:
Need help with these two implementations as well, and there's a second set of broken implementations of these functions in `validator_test.go`.

Reply:
The release notes [here](https://newreleases.io/project/github/kubernetes-sigs/controller-runtime/release/v0.15.0)

> Add client.{GroupVersionKindFor, IsObjectNamespaced} ([#2136](https://github.com/kubernetes-sigs/controller-runtime/pull/2136))
> 
>     The client.Client interface now requires and exposes these helper functions:
>         GroupVersionKindFor(Object) which returns the schema.GroupVersionKind for the object.
>         IsObjectNamespaced(Object) which returns true if the object's GroupVersionKind is namespaced, or false for global ones.

So I think we can follow suit of the other func and use:

```
func (c readOnlyClient) GroupVersionKindFor(obj runtime.Object) (schema.GroupVersionKind, error) {
	return c.client.GroupVersionKindFor(obj)
}
```

Reply:
Oh, that was easy, fixed! Thank you!